### DISCUSSION TITLE
Is it possible to install and configure presidio from a databricks environment without internet access?

### DISCUSSION THREAD

**Discussion by campbellwarren** [microsoft/presidio#1435] [Q&A] [answered]
Hi, I'm trying to configure presidio in a databricks environment with limited access to the internet.  When I do so it always attempts to download en_core_web_lg from the internet.  Even when I first dowload en_core_web_lg manually and load it to spacy, I always get an error after 
"_WARNING:presidio-analyzer:Model en_core_web_lg is not installed. Downloading..._".  
Even when I first load it to spacy using :
 _nlp = spacy.load("/tmp/en_core_web_lg/en_core_web_lg-3.7.1")_
I'm sure there is something I am doing wrong.  Any and all advice is greatly appreciated!

**Comment by campbellwarren** [comment, marked as answer]
In case someone else comes across this, I found the solution was to download the wheel file from https://spacy.io/models/en upload it to dbfs, then copy it from dbfs to the file system (dbutils.fs.cp). then install it with:
!pip install /tmp/en_core_web_lg-3.7.1-py3-none-any.whl
after that presidio no longer tires to download en_core_web_lg when envoked.

**Reply by omri374** [reply]
Exactly. Another thing to look out to is the URL recognizer which sometimes goes online to fetch some configuration, but this could be disabled if needed. See more here: https://github.com/microsoft/presidio/issues/1205

**Reply by campbellwarren** [reply]
Thank you, I'm sure it's a matter of time before I run into that problem :)
